diff -Naur -x '*~' -x '*.orig' -x '*.rej' bugzilla-4.0.1/Bugzilla/Install/Filesystem.pm bugzilla-4.0.1-dont-mess-file-perms/Bugzilla/Install/Filesystem.pm
--- bugzilla-4.0.1/Bugzilla/Install/Filesystem.pm	2011-04-28 03:34:25.000000000 +0200
+++ bugzilla-4.0.1-dont-mess-file-perms/Bugzilla/Install/Filesystem.pm	2011-06-16 14:04:14.000000000 +0200
@@ -145,19 +145,6 @@
         # Some .pl files are WS_EXECUTE because we want
         # users to be able to cron them or otherwise run
         # them as a secure user, like the webserver owner.
-        '*.cgi'           => { perms => WS_EXECUTE },
-        'whineatnews.pl'  => { perms => WS_EXECUTE },
-        'collectstats.pl' => { perms => WS_EXECUTE },
-        'importxml.pl'    => { perms => WS_EXECUTE },
-        'testserver.pl'   => { perms => WS_EXECUTE },
-        'whine.pl'        => { perms => WS_EXECUTE },
-        'email_in.pl'     => { perms => WS_EXECUTE },
-        'sanitycheck.pl'  => { perms => WS_EXECUTE },
-        'checksetup.pl'   => { perms => OWNER_EXECUTE },
-        'runtests.pl'     => { perms => OWNER_EXECUTE },
-        'jobqueue.pl'     => { perms => OWNER_EXECUTE },
-        'migrate.pl'      => { perms => OWNER_EXECUTE },
-        'install-module.pl' => { perms => OWNER_EXECUTE },
 
         'Bugzilla.pm'   => { perms => CGI_READ },
         "$localconfig*" => { perms => CGI_READ },
@@ -166,13 +153,6 @@
         'robots.txt'    => { perms => WS_SERVE },
         '.htaccess'     => { perms => WS_SERVE },
 
-        'contrib/README'       => { perms => OWNER_WRITE },
-        'contrib/*/README'     => { perms => OWNER_WRITE },
-        'docs/bugzilla.ent'    => { perms => OWNER_WRITE },
-        'docs/makedocs.pl'     => { perms => OWNER_EXECUTE },
-        'docs/style.css'       => { perms => WS_SERVE },
-        'docs/*/rel_notes.txt' => { perms => WS_SERVE },
-        'docs/*/README.docs'   => { perms => OWNER_WRITE },
         "$datadir/params"      => { perms => CGI_WRITE },
         "$datadir/old-params.txt"  => { perms => OWNER_WRITE },
         "$extensionsdir/create.pl" => { perms => OWNER_EXECUTE },
@@ -183,8 +163,6 @@
     # recurse through. These are directories we didn't create
     # in checkesetup.pl.
     my %non_recurse_dirs = (
-        '.'  => DIR_WS_SERVE,
-        docs => DIR_WS_SERVE,
     );
 
     # This sets the permissions for each item inside each of these 
@@ -205,54 +183,6 @@
          # Readable directories
          "$datadir/mining"     => { files => CGI_READ,
                                      dirs => DIR_CGI_READ },
-         "$libdir/Bugzilla"    => { files => CGI_READ,
-                                     dirs => DIR_CGI_READ },
-         $extlib               => { files => CGI_READ,
-                                     dirs => DIR_CGI_READ },
-         $templatedir          => { files => CGI_READ,
-                                     dirs => DIR_CGI_READ },
-         # Directories in the extensions/ dir are WS_SERVE so that
-         # the web/ directories can be served by the web server.
-         # But, for extra security, we deny direct webserver access to
-         # the lib/ and template/ directories of extensions.
-         $extensionsdir        => { files => CGI_READ,
-                                     dirs => DIR_WS_SERVE },
-         "$extensionsdir/*/lib" => { files => CGI_READ,
-                                      dirs => DIR_CGI_READ },
-         "$extensionsdir/*/template" => { files => CGI_READ,
-                                           dirs => DIR_CGI_READ },
-
-         # Content served directly by the webserver
-         images                => { files => WS_SERVE,
-                                     dirs => DIR_WS_SERVE },
-         js                    => { files => WS_SERVE,
-                                     dirs => DIR_WS_SERVE },
-         $skinsdir             => { files => WS_SERVE,
-                                     dirs => DIR_WS_SERVE },
-         'docs/*/html'         => { files => WS_SERVE,
-                                     dirs => DIR_WS_SERVE },
-         'docs/*/pdf'          => { files => WS_SERVE,
-                                     dirs => DIR_WS_SERVE },
-         'docs/*/txt'          => { files => WS_SERVE,
-                                     dirs => DIR_WS_SERVE },
-         'docs/*/images'       => { files => WS_SERVE,
-                                     dirs => DIR_WS_SERVE },
-         "$extensionsdir/*/web" => { files => WS_SERVE,
-                                     dirs => DIR_WS_SERVE },
-
-         # Directories only for the owner, not for the webserver.
-         '.bzr'                => { files => OWNER_WRITE,
-                                    dirs  => DIR_OWNER_WRITE },
-         t                     => { files => OWNER_WRITE,
-                                     dirs => DIR_OWNER_WRITE },
-         xt                    => { files => OWNER_WRITE,
-                                     dirs => DIR_OWNER_WRITE },
-         'docs/lib'            => { files => OWNER_WRITE,
-                                     dirs => DIR_OWNER_WRITE },
-         'docs/*/xml'          => { files => OWNER_WRITE,
-                                     dirs => DIR_OWNER_WRITE },
-         'contrib'             => { files => OWNER_EXECUTE,
-                                     dirs => DIR_OWNER_WRITE, },
     );
 
     # --- FILES TO CREATE --- #
@@ -265,14 +195,10 @@
         # Directories that are read-only for cgi scripts
         "$datadir/mining"       => DIR_CGI_READ,
         "$datadir/extensions"   => DIR_CGI_READ,
-        $extensionsdir          => DIR_CGI_READ,
         # Directories that cgi scripts can write to.
         $attachdir              => DIR_CGI_WRITE,
         $graphsdir              => DIR_CGI_WRITE | DIR_ALSO_WS_SERVE,
         $webdotdir              => DIR_CGI_WRITE | DIR_ALSO_WS_SERVE,
-        # Directories that contain content served directly by the web server.
-        "$skinsdir/custom"      => DIR_WS_SERVE,
-        "$skinsdir/contrib"     => DIR_WS_SERVE,
     );
 
     # The name of each file, pointing at its default permissions and
